File: clash.php
<?php

// Author: Quasilyte

exec('date +%s%N > buf/ts1 2>&1');

$map[0] = explode("\n", file_get_contents('input/file1.txt'));
unset($map[0][count($map[0]) - 1]);
$map[1] = explode("\n", file_get_contents('input/file2.txt'));
unset($map[1][count($map[1]) - 1]);

$spell = '';
$cs = 0;
$vowels = ['a' => 1, 'e' => 1, 'i' => 1, 'o' => 1, 'u' => 1];

$maxLen = 0;
$minLen = PHP_INT_MAX;

foreach($map[0] as $word) {
  if(($len = strlen($word)) > $maxLen) {
    $maxLen = $len;
  } elseif($len < $minLen) {
    $minLen = $len;
  }

  // Составляем заклинание.
  if(($ch = ord($word[0])) >= 97 && $ch <= 101) {
    $spell .= $word . '_';

    // Считаем гласные.
    for($i = 0, $len = strlen($word); $i < $len; ++$i) {
      if(isset($vowels[$word[$i]])) {
        $cs += 1;
      }
    }
  }
}

for($i = 0, $sum = 0; $i < 250; ++$i) {
  $sum += sqrt($map[1][$i] * $map[1][$i + 250]);
}

function acker($a, $b) {
  if(!$a) {
    return ++$b;
  } elseif($a && !$b) {
    return acker(--$a, 1);
  } elseif($a && $b) {
    return acker($a - 1, acker($a, --$b));
  }
}

$ackermann = acker(3, 3);

exec('date +%s%N > buf/ts2 2>&1');
$ts = (file_get_contents('buf/ts2') - file_get_contents('buf/ts1')) / 100000;

################################################################################

echo "elapsed: $ts\n";
echo "max length: $maxLen, min length: $minLen\n";
echo "spell length: " . strlen($spell) . ", checksum: $cs\n";
echo "sum: $sum\n";
echo "ackermann: $ackermann\n";

File: clash.pl
# Author: Quasilyte

`date +%s%N > buf/ts1 2>&1`;

sub fslurp {
  local $/ = undef;
  open FILE, $_[0];

  my $rbuf = <FILE>;
  close FILE;
  return $rbuf;
}

my @map = (
  [split /\n/, fslurp('input/file1.txt')],
  [split /\n/, fslurp('input/file2.txt')]
);

my $spell = '', $cs = 0, $lmax = 0, $lmin = 255;
my $sum = 0;

for(@{$map[0]}) {
  my $len = length;

  if($len > $lmax) {
    $lmax = $len;
  } elsif($len < $lmin) {
    $lmin = $len;
  }

  if(ord() >= 97 && ord() <= 101) {
    $spell .= $_ . '_';
  }
}

$cs = ($spell =~ tr/aeiou//);

for(my $i = 0; $i < 250; ++$i) {
  $sum += sqrt @{$map[1]}[$i] * @{$map[1]}[$i + 250];
}

sub acker {
  if(!$_[0]) {
    return $_[1] + 1;
  } elsif($_[0] && !$_[1]) {
    return acker($_[0] - 1, 1);
  } elsif($_[0] && $_[1]) {
    return acker($_[0] - 1, acker($_[0], $_[1] - 1));
  }
}

my $ackermann = acker(3, 3);

`date +%s%N > buf/ts2 2>&1`;
my $ts = ((fslurp 'buf/ts2') - (fslurp 'buf/ts1')) / 100000;

################################################################################

print "elapsed: $ts\n";
print "max length: $lmax, min length: $lmin\n";
print "spell length: " . length $spell;
print ", checksum: $cs\n";
print "sum: $sum\n";
print "ackermann: $ackermann\n";
